### PR TITLE
Fix RegionInfo creation when using LOCALE_CUSTOM_UNSPECIFIED LCID

### LINKED ISSUE #64893: RegionInfo(4096) behaves differently depending on preceding method calls
### Description

The documentation for [`RegionInfo(int culture)`](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.regioninfo.-ctor?view=net-6.0#system-globalization-regioninfo-ctor(system-int32)) says it will throw an `ArgumentException` when "`culture` specifies either an invariant, custom, or neutral culture."

However, when passed a value of 4096 (LOCALE_CUSTOM_UNSPECIFIED) the constructor behaves differently depending on preceding method calls.

### Reproduction Steps

```cs
using System;
using System.Globalization;
using System.Linq;

if (args.Contains("get-cultures"))
{
    CultureInfo.GetCultures(CultureTypes.SpecificCultures);
}

if (args.Contains("ai"))
{
    new RegionInfo("ai");
}

Console.WriteLine(new RegionInfo(4096).EnglishName);
```

### Expected behavior

I expect `new RegionInfo(4096)` to throw an `ArgumentException`, regardless of the preceding method calls.

### Actual behavior

On my machine:

If run without arguments, the program prints "Australia".

If run with the argument "get-cultures", the program prints "Ethiopia".

If run with the argument "ai", the program throws an `ArgumentException` with the message "Culture ID 4096 (0x1000) is a neutral culture; a region cannot be created from it. (Parameter 'culture')".

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

I have tested this on .NET 5 and .NET 6.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/64893

Don't allow creating RegionInfo when using LOCALE_CUSTOM_UNSPECIFIED (0x1000) Lcid.

### CODE REVIEW

File: src/libraries/System.Globalization/tests/System/Globalization/RegionInfoTests.cs [hunk 1]
```diff
@@ -254,5 +254,18 @@ public void EqualsTest(RegionInfo regionInfo1, object obj, bool expected)
                 Assert.Equal(expected, regionInfo1.GetHashCode().Equals(obj.GetHashCode()));
             }
         }
+
+        [Fact]
+        public void ValidateThrowingWhenUsingCustomUnspecifiedLcid()
+        {
+            // When trying to create RegionInfo using Lcid, we call LcidToLocaleName to map the Lcid to the culture name.
+            // LOCALE_CUSTOM_UNSPECIFIED (0x1000) is considered invalid Lcid to create cultures with. Sometime Windows can
```
Comment (safern) [inline src/libraries/System.Globalization/tests/System/Globalization/RegionInfoTests.cs:262]:
```suggestion
            // LOCALE_CUSTOM_UNSPECIFIED (0x1000) is considered invalid Lcid to create cultures with. Sometimes Windows can
```

File: src/libraries/System.Private.CoreLib/src/System/Globalization/RegionInfo.cs [hunk 2]
```diff
@@ -63,7 +63,7 @@ public RegionInfo(int culture)
                 throw new ArgumentException(SR.Format(SR.Argument_CultureIsNeutral, culture), nameof(culture));
             }
 
-            if (culture == CultureInfo.LOCALE_CUSTOM_DEFAULT)
+            if (culture == CultureInfo.LOCALE_CUSTOM_DEFAULT || culture == CultureInfo.LOCALE_CUSTOM_UNSPECIFIED)
             {
                 // Not supposed to be neutral
                 throw new ArgumentException(SR.Format(SR.Argument_CustomCultureCannotBePassedByNumber, culture), nameof(culture));
```
Comment (safern) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/RegionInfo.cs:69]: Is this the right exception message for `LOCALE_CUSTOM_UNSPECIFIED`? 

https://github.com/dotnet/runtime/blob/3ae87395f638a533f37b8e3385f6d3f199a72f4f/src/libraries/System.Private.CoreLib/src/Resources/Strings.resx#L970

Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/RegionInfo.cs:69]: Yes, it still correct exception message.

**Review by safern — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs [hunk 3]
```diff
@@ -186,6 +186,11 @@ internal static unsafe CultureData GetCurrentRegionData()
         {
             Debug.Assert(!GlobalizationMode.Invariant);
 
+            if (culture == CultureInfo.LOCALE_CUSTOM_UNSPECIFIED)
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs:189]: Worth adding a comment?

File: src/libraries/System.Globalization/tests/System/Globalization/RegionInfoTests.cs [hunk 4]
```diff
@@ -254,5 +254,18 @@ public void EqualsTest(RegionInfo regionInfo1, object obj, bool expected)
...
+        public void ValidateThrowingWhenUsingCustomUnspecifiedLcid()
+        {
+            // When trying to create RegionInfo using Lcid, we call LcidToLocaleName to map the Lcid to the culture name.
+            // LOCALE_CUSTOM_UNSPECIFIED (0x1000) is considered invalid Lcid to create cultures with. Sometimes Windows can
+            // map which is wrong. This happen if we enumerate the cultures before trying to create the RegionInfo object.
+            // This test is to ensure we'll not allow this creation in .NET.
+
+            CultureInfo [] cultures = CultureInfo.GetCultures(CultureTypes.SpecificCultures);
+
+            AssertExtensions.Throws<ArgumentException>(() => new RegionInfo(4096));
```
Comment (stephentoub) [inline src/libraries/System.Globalization/tests/System/Globalization/RegionInfoTests.cs:268]: Is there an argument name included in the exception that should be verified?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs [hunk 5]
```diff
@@ -186,6 +186,13 @@ internal static unsafe CultureData GetCurrentRegionData()
         {
             Debug.Assert(!GlobalizationMode.Invariant);
 
+            // LOCALE_CUSTOM_UNSPECIFIED ia an unspecified custom locale, used to identify all supplemental locales.
```
Comment (canton7) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Windows.cs:189]: "ia an"

**Pull request merged**